assets/mechanics.js

```diff
@@ -105,18 +108,23 @@ function update() {
     preventGoingThroughBottom(); // conditions wen ball hit bottem of canvis
     /// only needed for ground
     floors.forEach((element) => {
-      element.floorcollision(); // these condition make sure thea are repeated in this function
-      //   console.log(obstacle);
-      element.movefloorUp();
+
       
-      if (element.floorY < 80) {
-        floors.shift()
+      if (element.floorY < 80 && element.deleted == false) {
+        // We should remove this floor.
+        element.deleted = true
         floorCount += 1 
         if ( floorCount % 2 == 0){
             createObjects(1);
         }
       }
+      if (element.deleted == false){
```

Comment:
We only redraw and do cool things with elements which have not been deleted yet.

All of this is still in a list (Which is terrible! We don't remove these unused objects... they just clog up memory)

This would be a problem if you kept this game running for a few hours or more.

Reply:
i don't know wen i do this it crashes immidiatly